debezium-core/src/main/java/io/debezium/relational/topic/ByLogicalTableTopicMapper.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.relational.topic;
+
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+
+import io.debezium.relational.Table;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * A logical table consists of one or more physical tables with the same schema. A common use case is sharding -- the
+ * two physical tables `db_shard1.my_table` and `db_shard2.my_table` together form one logical table.
+ *
+ * This TopicMapper allows us to send change events from both physical tables to one topic.
+ *
+ * @author David Leibovic
+ */
+public class ByLogicalTableTopicMapper extends TopicMapper {
+
+    public String getTopicName(String topicPrefix, Table table) {
+        final String fullyQualifiedTableName = composeFullyQualifiedTableName(topicPrefix, table);
+        Pattern logicalTablePattern = Pattern.compile("^.*?(?=\\..+\\..+)\\.(?<logicalDb>etsy_.+?(?=(_\\d+\\.)|\\.))(_\\d+)?\\.(?<table>.+)$");
+        Matcher logicalTableMatcher = logicalTablePattern.matcher(fullyQualifiedTableName);
+        if (logicalTableMatcher.matches()) {
+            return logicalTableMatcher.replaceAll("${logicalDb}.${table}");
```

Comment:
In case it wasn't obvious (hah), this crazy regex and corresponding replacement pattern has the effect of applying the following transformations:
```
shard001.etsy.com:8888.etsy_shard_001.my_table -> etsy_shard.my_table
dbindex.etsy.com:8888.etsy_index.my_other_table -> etsy_index.my_other_table
```